crates/trie/trie/src/trie_cursor/mod.rs

```diff
@@ -52,3 +53,29 @@ pub trait TrieCursor: Send + Sync {
     /// Get the current entry.
     fn current(&mut self) -> Result<Option<Nibbles>, DatabaseError>;
 }
+
+/// A trait for iterating over a range of block numbers.
+#[auto_impl::auto_impl(&mut, Box)]
+pub trait TrieRangeWalker<V>: Send + Sync {
```

Comment:
nit: can make it an associated type similar to other cursor implementations
```suggestion
pub trait TrieRangeWalker: Send + Sync {
    type Value;
```

Reply:
Makes sense to use the associated type instead of generic here since the implementation is tied to one concrete type of `Value`.